packages/flutter_tools/lib/src/migrate/migrate_utils.dart

```diff
@@ -0,0 +1,325 @@
+    final ProcessResult result = await Process.run('git', cmdArgs);
+    checkForErrors(result, logger, allowedExitCodes: <int>[-1], commandDescription: 'git ${cmdArgs.join(' ')}');
+    return MergeResult(result, localPath);
+  }
+
+  /// Calls `git init` on the workingDirectory.
+  static Future<void> gitInit(String workingDirectory, Logger logger) async {
+    final List<String> cmdArgs = <String>['init'];
+    final ProcessResult result = await Process.run('git', cmdArgs, workingDirectory: workingDirectory);
+    checkForErrors(result, logger, allowedExitCodes: <int>[0], commandDescription: 'git ${cmdArgs.join(' ')}');
+  }
+
+  /// Returns true if the workingDirectory git repo has any uncommited changes.
+  static Future<bool> hasUncommitedChanges(String workingDirectory, Logger logger) async {
+    final List<String> cmdArgs = <String>['diff', '--quiet', 'HEAD', '--', '.', "':(exclude)$kDefaultMigrateWorkingDirectoryName'"];
+    final ProcessResult result = await Process.run('git', cmdArgs, workingDirectory: workingDirectory);
+    checkForErrors(result, logger, allowedExitCodes: <int>[-1], commandDescription: 'git ${cmdArgs.join(' ')}');
+    if (result.exitCode == 0) {
+      return false;
+    }
+    return true;
+  }
+
+  /// Returns true if the workingDirectory is a git repo.
+  static Future<bool> isGitRepo(String workingDirectory, Logger logger) async {
+    final List<String> cmdArgs = <String>['rev-parse', '--is-inside-work-tree'];
+    final ProcessResult result = await Process.run('git', cmdArgs, workingDirectory: workingDirectory);
+    checkForErrors(result, logger, allowedExitCodes: <int>[-1], commandDescription: 'git ${cmdArgs.join(' ')}');
+    if (result.exitCode == 0) {
+      return true;
+    }
+    return false;
+  }
+
+  /// Returns true if the file at `filePath` is covered by the `.gitignore`
+  static Future<bool> isGitIgnored(String filePath, String workingDirectory, Logger logger) async {
+    final List<String> cmdArgs = <String>['check-ignore', filePath];
+    final ProcessResult result = await Process.run('git', cmdArgs, workingDirectory: workingDirectory);
+    checkForErrors(result, logger, allowedExitCodes: <int>[0, 1, 128], commandDescription: 'git ${cmdArgs.join(' ')}');
+    return result.exitCode == 0;
+  }
+
+  /// Runs `flutter pub upgrate --major-revisions`.
+  static Future<void> flutterPubUpgrade(String workingDirectory, Logger logger) async {
+    final List<String> cmdArgs = <String>['pub', 'upgrade', '--major-versions'];
+    final ProcessResult result = await Process.run('flutter', cmdArgs, workingDirectory: workingDirectory);
+    checkForErrors(result, logger, allowedExitCodes: <int>[0], commandDescription: 'flutter ${cmdArgs.join(' ')}');
+  }
+
+  /// Runs `./gradlew tasks` in the android directory of a flutter project.
+  static Future<void> gradlewTasks(String workingDirectory, Logger logger) async {
+    final String baseCommand = Platform.isWindows ? 'gradlew.bat' : './gradlew';
```

Comment:
rather than `io.Platform`, use `package:platform` and direct inject in the `MigrateUtils` constructor.